aspnetcore/grpc/clientfactory.md

```diff
@@ -0,0 +1,115 @@
+---
+title: gRPC client factory integration in .NET Core
+author: jamesnk
+description: Learn how to create gRPC clients using the client factory.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: jamesnk
+ms.date: 08/06/2019
+uid: grpc/clientfactory
+---
+# gRPC client factory integration in .NET Core
+
+gRPC integration with HttpClientFactory offers a centralized way to create gRPC clients. It can be used as an alternative to [configuring stand-alone gRPC client instances](xref:grpc/client). Factory integration is available in the [Grpc.Net.ClientFactory](https://www.nuget.org/packages/Grpc.Net.ClientFactory) NuGet package.
+
+The factory offers the following benefits:
+
+* Provides a central location for configuring logical gRPC client instances
+* Manages the lifetime of the underlying `HttpClientMessageHandler`
+* Automatic propagation of deadline and cancellation in a ASP.NET Core gRPC service
+
+## Register gRPC clients
+
+To register a gRPC client, the generic `AddGrpcClient` extension method can be used within `Startup.ConfigureServices`, specifying the gRPC typed client class and service address:
+
+```csharp
+services.AddGrpcClient<Greeter.GreeterClient>(o =>
+{
+    o.BaseAddress = new Uri("http://localhost:5001");
+});
+```
+
+The gRPC client type is registered as transient with DI. The client can now be injected and consumed directly in types create by DI. ASP.NET Core MVC controllers, SignalR hubs and gRPC services are all places where gRPC clients can automatically be injected:
+
+```csharp
+public class AggregatorService : Aggregator.AggregatorBase
+{
+    private readonly Greeter.GreeterClient _client;
+
+    public AggregatorService(Greeter.GreeterClient client)
+    {
+        _client = client;
+    }
+
+    public override async Task SayHellos(HelloRequest request,
```

Comment:
Should this method be called `SayHello` instead?

Reply:
No, it is streaming multiple hellos.